Bug: (Unable to resolve path to module '@eslint/compat.'.eslintimport/no-unresolved)

### Which packages are affected?

- [X] `@eslint/compat`
- [ ] `@eslint/config-array`
- [ ] `@eslint/migrate-config`
- [ ] `@eslint/object-schema`

### Environment

Node version: v18.20.3
pnpm version: 9.1.4
ESLint version: 9.4.0
Operating System: MacOS 11.5.2
VS Code

### What did you do?

My project is written in JavaScript, and I would like to add ESLint to it.

### What did you expect to happen?

`import { fixupConfigRules } from '@eslint/compat';`
This line should not be prompted with an error

### What actually happened?

Unable to resolve path to module '@eslint/compat.'.eslint[import/no-unresolved](https://github.com/import-js/eslint-plugin-import/blob/v2.29.1/docs/rules/no-unresolved.md)

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/vitejs-vite-bm5pcp

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

<img width="908" alt="image" src="https://github.com/eslint/rewrite/assets/3996569/c61683eb-841a-430b-a2d6-25a98d8cb483">

When I added "main": "./dist/esm/index.js" to the package.json file under the node_modules/@eslint/compact directory, my editor stopped showing errors.

> The reproduction case you linked to does not show this error. Please provide a repro case where we can see the error in action and we can take a look.

Hi @nzakas, I could reproduce this. This is related to the `no-unresolved`  rule from `eslint-plugin-import`, but I'm not sure if it is because `@eslint/compat` does not have a `main` property in its `package.json`. The rule does not complain when there is a `main` property.

```bash
npx eslint .

/home/projects/vitejs-vite-bm5pcp/eslint.config.js
   3:34  error  Unable to resolve path to module '@eslint/compat'  import/no-unresolved
```

--- a/packages/compat/package.json
+++ b/packages/compat/package.json
@@ -3,6 +3,8 @@
   "version": "1.0.3",
   "description": "Compatibility utilities for ESLint",
   "type": "module",
+  "main": "dist/esm/index.js",
+  "types": "dist/esm/index.d.ts",
   "exports": {
     "require": {
       "types": "./dist/cjs/index.d.cts",

--- a/packages/config-array/package.json
+++ b/packages/config-array/package.json
@@ -4,6 +4,8 @@
   "description": "General purpose glob-based configuration matching.",
   "author": "Nicholas C. Zakas",
   "type": "module",
+  "main": "dist/esm/index.js",
+  "types": "dist/esm/index.d.ts",
   "exports": {
     "require": {
       "types": "./dist/cjs/index.d.cts",

--- a/packages/object-schema/package.json
+++ b/packages/object-schema/package.json
@@ -3,6 +3,8 @@
   "version": "2.1.3",
   "description": "An object schema merger/validator",
   "type": "module",
+  "main": "dist/esm/index.js",
+  "types": "dist/esm/index.d.ts",
   "exports": {
     "require": {
       "types": "./dist/cjs/index.d.cts",

